src/main/java/fr/xephi/authme/security/totp/TotpAuthenticator.java

```diff
@@ -0,0 +1,74 @@
+package fr.xephi.authme.security.totp;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.warrenstrange.googleauth.GoogleAuthenticator;
+import com.warrenstrange.googleauth.GoogleAuthenticatorKey;
+import com.warrenstrange.googleauth.GoogleAuthenticatorQRGenerator;
+import com.warrenstrange.googleauth.IGoogleAuthenticator;
+import fr.xephi.authme.service.BukkitService;
+import org.bukkit.entity.Player;
+
+import javax.inject.Inject;
+
+/**
+ * Provides rudimentary TOTP functions (wraps third-party TOTP implementation).
+ */
+public class TotpAuthenticator {
+
+    private final IGoogleAuthenticator authenticator;
+    private final BukkitService bukkitService;
+
+
+    @Inject
+    TotpAuthenticator(BukkitService bukkitService) {
+        this(new GoogleAuthenticator(), bukkitService);
```

Comment:
This creates by default Base32 keys with SHA1. Since SHA1 is broken we should use something stronger or do you think it's not relevant in this case?

Reply:
I can try to change it and I'll check if Google Authenticator still works with it. As long as it does it's definitely a great improvement. Thanks for noticing!